packages/datadog_tracking_http_client/lib/src/tracking_http.dart

```diff
@@ -0,0 +1,204 @@
+/// Spans sent), see [DdSdkConfiguration.firstPartyHosts]. You can also set
+/// first party hosts after initialization setting [DatadogSdk.firstPartyHosts]
+///
+/// DatadogClient only modifies calls made through itself, unlike
+/// [DatadogTrackingHttpClient], which overrides all calls made by [HttpClient]
+/// and includes network calls made by Flutter. However, DatadogClient allows
+/// you to compose with other [http.BaseClient] based libraries like
+/// `cupertino_http` and `cronet_http`, which [DatadogTrackingHttpClient] would
+/// miss.
+///
+/// DatadogClient and [DatadogTrackingHttpClient] can be used together if needed,
+/// and will not interfere with each other.
+///
+/// See also [DatadogTrackingHttpClient]
+class DatadogClient extends http.BaseClient {
+  final DatadogSdk datadogSdk;
+  final Uuid _uuid = const Uuid();
+  final http.Client _innerClient;
+
+  DatadogClient({
+    required this.datadogSdk,
+    http.Client? innerClient,
+  }) : _innerClient = innerClient ?? http.Client();
+
+  @override
+  Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    final rum = datadogSdk.rum;
+    // Never track on web
+    if (kIsWeb || rum == null) {
+      return _innerClient.send(request);
+    }
+
+    return _trackingSend(request, rum);
+  }
+
+  Future<http.StreamedResponse> _trackingSend(
+      http.BaseRequest request, DdRum rum) async {
+    String? rumKey;
+    bool isFirstParty = false;
+
+    try {
+      isFirstParty = datadogSdk.isFirstPartyHost(request.url);
+      rumKey = _uuid.v1();
+      bool shouldSample = isFirstParty && rum.shouldSampleTrace();
+      final attributes =
+          _appendRequestHeaders(request, rumKey, isFirstParty, shouldSample);
+
+      final rumHttpMethod = rumMethodFromMethodString(request.method);
+      rum.startResourceLoading(
+          rumKey, rumHttpMethod, request.url.toString(), attributes);
+    } catch (e, st) {
+      datadogSdk.internalLogger.sendToDatadog(
+        '$DatadogClient encountered an error while attempting '
+        ' to track a send call: $e',
+        st,
+        e.runtimeType.toString(),
+      );
+      // Since there was an error, don't attempt any more tracking
+      rumKey = null;
+    }
+
+    http.StreamedResponse response;
+    try {
+      response = await _innerClient.send(request);
+    } catch (e) {
+      if (rumKey != null) {
+        try {
+          rum.stopResourceLoadingWithErrorInfo(
+              rumKey, e.toString(), e.runtimeType.toString());
+        } catch (innerE, st) {
+          datadogSdk.internalLogger.sendToDatadog(
+            '$DatadogClient encountered an error while attempting '
+            ' to track a send - error: $e',
+            st,
+            e.runtimeType.toString(),
+          );
+          rumKey = null;
+        }
+      }
+
+      rethrow;
+    }
+
+    if (rumKey != null) {
+      try {
+        // Copy the response so we can spy on the stream. Use
```

Comment:
Is some sentence not finished here? Or it is ok?

Reply:
Does appear there's an extra word but I don't think there was more to say.